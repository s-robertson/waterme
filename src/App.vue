<template>
  <div id="app">
    <div v-if="loading" class="app_loader">
      Loading...
    </div>
    <div v-else class="app__container">
      <AppHeader />
      <div class="app__content">
        <router-view />
      </div>
      <AppFooter />
    </div>
  </div>
</template>

<script>
import { mapState, mapActions } from "vuex";
import AppHeader from "@/components/AppHeader";
import AppFooter from "@/components/AppFooter";
export default {
  computed: {
    ...mapState(["loading"])
  },
  methods: {
    ...mapActions(["init"])
  },
  mounted() {
    this.init();
  },
  components: { AppFooter, AppHeader }
};
</script>

<style lang="scss">
@import "./assets/variables";
@import "./assets/app.scss";

#app {
  min-height: 100%;
  position: relative;
}

.app__content {
  margin: 2rem auto 0 auto;
  padding-bottom: 4rem;
  max-width: $app-max-width;
}
</style>
