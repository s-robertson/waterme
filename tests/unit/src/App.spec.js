import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import Vuex from "vuex";
import App from "@/App";
import { createStore } from "../helpers";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe("App Component", () => {
  const defaultStoreOptions = {
    state: {
      loading: false
    },
    actions: {
      init: jest.fn()
    }
  };

  const router = new VueRouter();

  it("should render correctly", () => {
    const wrapper = shallowMount(App, {
      store: createStore(defaultStoreOptions),
      localVue,
      router
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should display a loading message while the app is loading", () => {
    const wrapper = shallowMount(App, {
      store: createStore(defaultStoreOptions, { state: { loading: true } }),
      localVue,
      router
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
