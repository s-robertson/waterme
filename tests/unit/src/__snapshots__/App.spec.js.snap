// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should display a loading message while the app is loading 1`] = `
<div id="app">
  <div class="app__loader">
    Loading...
  </div>
</div>
`;

exports[`App Component should render correctly 1`] = `
<div id="app">
  <div class="app__container">
    <appheader-stub></appheader-stub>
    <div class="app__content">
      <router-view-stub name="default"></router-view-stub>
    </div>
    <appfooter-stub></appfooter-stub>
  </div>
</div>
`;
